### ISSUE TITLE
Allow for custom explanations when using SelectorFormat regexps

### ISSUE THREAD

**Issue by giuseppeg** [brigade/scss-lint#368]
When using regular expressions to validate custom conventions (selectors) the current linter error sometimes is not that human readable due to the complexity of the regexp.
eg.
```perl
should match regex /^[a-z]([-]?[a-z0-9]+)*(__[a-z0-9]([-]?[a-z0-9]+)*)?((_[a-z0-9]([-]?[a-z0-9]+)*){2})?$/
```

It would be nice to be able to set custom linter error messages for `#{type}_convention` and `convention` and possibly provide a way to add a placeholder that will later be replaced with the regexp.

The implementation is pretty straightforward. lib/scss_lint/linter/selector_format.rb:78

```ruby
convention = CONVENTIONS[convention_name] || {
  explanation: (
                  config["#{type}_explanation"] ||
                  config['explanation'] ||
                  # the default msg should probably go outside
                  # and we should gsub only the config ones
                  "should match regex /#{convention_name}/"
              ).gsub('#{convention_name}', "/#{convention_name}/"),
  validator: ->(name) { name =~ /#{convention_name}/ }
}
```

We could then set a custom message in `.scss-lint.yml`

```yaml
SelectorFormat:
  enabled: true
  class_convention: ^(foo|bar)
  class_explanation: "Sorry buddy, your selector is wrong. Read more @ http://scss.docs/linter or change it so it matches #{convention_name}"
```

We need this @Yelp but I believe that it would be useful to a whole bunch of other people :)

I may submit a PR if you guys are ok with adding this feature.

Cheers

**Comment by lencioni** [comment]
Thanks @giuseppeg for the interest in scss-lint! This makes sense to me, but @sds really has the most context here so it would be best for him to chime in about this. I believe he will be unavailable at least until the weekend.

**Comment by sds** [comment]
Hey @giuseppeg,

This feature as you describe seems fine. I'd be happy to merge a pull request with tests!

**Comment by giuseppeg** [comment]
@sds @lencioni awesome thanks!

**Comment by giuseppeg** [comment]
@sds thank you for taking the time to implementing this <3
